pkg/controller/replication/replication_controller.go

```diff
@@ -245,88 +314,104 @@ func (r *ReconcileReplication) ensureReplication(ctx context.Context, harborClie
 		DestNamespace: originalReplication.Spec.DestNamespace,
 		Override:      originalReplication.Spec.Override,
 		Enabled:       originalReplication.Spec.Enabled,
 		Deletion:      originalReplication.Spec.Deletion,
 	}
 
-	if originalReplication.Spec.Filters != nil {
-		hf := []*h.Filter{}
-		for _, v := range originalReplication.Spec.Filters {
-			err := internal.CheckFilterType(v.Type)
-			if err != nil {
-				return h.ReplicationPolicy{}, err
-			}
-			hf = append(hf, &h.Filter{
-				Type:  v.Type,
-				Value: v.Value,
-			})
-		}
-		newRep.Filters = hf
+	filters, err := addReplicationFilters(originalReplication.Spec.Filters)
+	if err != nil {
+		return nil, err
 	}
 
+	newRep.Filters = filters
+
 	if originalReplication.Spec.Trigger != nil {
-		ht := &h.Trigger{}
-		validatedType, err := internal.CheckAndGetReplicationTriggerType(originalReplication.Spec.Trigger.Type)
+		triggerType, err := enumReplicationTrigger(originalReplication.Spec.Trigger.Type)
 		if err != nil {
-			return h.ReplicationPolicy{}, err
+			return nil, err
 		}
 
-		ht.Type = validatedType
-		if originalReplication.Spec.Trigger.Settings != nil {
-			ht.Settings = &h.TriggerSettings{Cron: originalReplication.Spec.Trigger.Settings.Cron}
+		newRep.Trigger = &modelv1.ReplicationTrigger{
+			TriggerSettings: &modelv1.TriggerSettings{
+				Cron: originalReplication.Spec.Trigger.Settings.Cron,
+			},
+			Type: triggerType,
 		}
-		newRep.Trigger = ht
 	}
 
 	if originalReplication.Spec.SrcRegistry != nil && originalReplication.Spec.DestRegistry != nil {
-		return h.ReplicationPolicy{}, fmt.Errorf("both dest_registry and src_registry are set! Please specify only one of them")
+		return &modelv1.ReplicationPolicy{},
+			fmt.Errorf("both dest_registry and src_registry are set! Please specify only one of them")
 	}
+
 	if originalReplication.Spec.SrcRegistry != nil {
-		hReg, err := r.getHarborRegistryFromRef(context.Background(), originalReplication.Spec.SrcRegistry, originalReplication.Namespace)
+		hReg, err := r.getHarborRegistryFromRef(context.Background(),
+			originalReplication.Spec.SrcRegistry,
+			originalReplication.Namespace)
 		if err != nil {
-			return h.ReplicationPolicy{}, err
+			return &modelv1.ReplicationPolicy{}, err
 		}
+
 		newRep.SrcRegistry = hReg
 	} else if originalReplication.Spec.DestRegistry != nil {
-		hReg, err := r.getHarborRegistryFromRef(context.Background(), originalReplication.Spec.DestRegistry, originalReplication.Namespace)
+		hReg, err := r.getHarborRegistryFromRef(context.Background(),
+			originalReplication.Spec.DestRegistry,
+			originalReplication.Namespace)
 		if err != nil {
-			return h.ReplicationPolicy{}, err
+			return &modelv1.ReplicationPolicy{}, err
 		}
+
 		newRep.DestRegistry = hReg
 	}
 
-	return newRep, nil
+	return &newRep, nil
+}
+
+func addReplicationFilters(originalFilters []registriesv1alpha1.ReplicationFilter) (newFilters []*modelv1.
+	ReplicationFilter, err error) {
```

Comment:
In general, I find the formatting in some cases... odd (in this case, the line break admist the return type list). Was this `gofmt`, or some other formatting tool?

Reply:
Yup, to be exact: in this case it was GoLand's "hard wrap" setting, which broke lines at the `.` operator.
Changing it to break lines at the opening parenthesis of `(newFilters [...])` like below:
```
func addReplicationFilters(originalFilters []registriesv1alpha1.ReplicationFilter) (
	newFilters []*modelv1.ReplicationFilter, err error) {
```